support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java

```diff
@@ -296,4 +313,62 @@ private Object convert(String name, ParameterValue v) throws IOException, Interr
     }
 
     private static final long serialVersionUID = 1L;
+    
+    // TODO: replace with a call to MultipartFormDataParser.isMultipart once available in parent core
+    private static boolean isMultipart(StaplerRequest request) {
+        if (request == null) {
+            return false;
+        }
+
+        String contentType = request.getContentType();
+        if (contentType == null) {
+            return false;
+        }
+
+        String[] parts = contentType.split(";");
+        if (parts.length == 0) {
+            return false;
+        }
+
+        for (int i = 0; i < parts.length; i++) {
+            if ("multipart/form-data".equals(parts[i])) {
+                return true;
+            }
+        }
+
+        return false;
+    }    
+    
+    // TODO: total hack to gain access to all parsedFormData from Stapler RequestImpl
+    // It sooo sucks that RequestImpl makes the decision it does wrt isFormField ... why not let the 
+    // caller do that kind of filtering if they need it !!    
+    private static FileItem getFileItem(StaplerRequest request, String name) throws ServletException, IOException {
+        FileItem item = request.getFileItem(name);
+        
+        if (item != null) {
+            return item;
+        }
+        if (!(request instanceof RequestImpl)) {
+            return null;
+        }
+
+        try {
+            Field parsedFormDataField = request.getClass().getDeclaredField("parsedFormData");
```

Comment:
@jglick Right .. I added the earlier explicit check for `RequestImpl` after that `request.getClass()` was added ... should have changed that at the same time.

Better again though would be to not have to do shit like this at all ;) I doubt we ever do hacks like this, or is it acceptable as a stopgap while we wait for changes to come through from Stapler i.e. for me to make a change there to make it possible to get the proper `FileItem` from `StaplerRequest` (not done, but will do if we think this is how to go).